### PR TITLE
Integrating B2C mpesa API using Node.js

### PR DESCRIPTION
**Pull request by Roba-W**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by Roba-W** [comment]
This fixes #1950

File: articles/integrating-b2c-mpesa-api-using-node-js/index.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
+### Integrating B2C Mpesa API using Node.js
+
+Mpesa can be described as a phone-based mobile banking service. Since its inauguration in 2007 by Vodafone Group PLC and Safaricom in Kenya, it has expanded to eight more countries in Africa.
```
Comment (WanjaMIKE) [inline articles/integrating-b2c-mpesa-api-using-node-js/index.md:3]: Change `M-Pesa` spelling throughout the paper

File: articles/integrating-b2c-mpesa-api-using-node-js/index.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+An API (Application Programming Interface) is an intermediary between two software applications that enables them to communicate.
+
+B2C Mpesa API enables software developers to integrate the functionality of merchants sending payments to their customers in their applications.
```
Comment (WanjaMIKE) [inline articles/integrating-b2c-mpesa-api-using-node-js/index.md:7]: Could you tell the readers that B2C stands for `Business to Consumer`

File: articles/integrating-b2c-mpesa-api-using-node-js/index.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+In this article, we will integrate the B2C Mpesa API to a Node.js RESTful API. We will consume the API using [swagger](https://swagger.io/).
+
+### Prerequisites
+
+To follow along in this article, it is helpful to have the following:
+
+- [Node.js](https://nodejs.org/en/) installed on your computer.
+- Some basic knowledge working with JavaScript.
+- A text editor installed.
```
Comment (WanjaMIKE) [inline articles/integrating-b2c-mpesa-api-using-node-js/index.md:19]: Inform the reader what text editor you will be using

File: articles/integrating-b2c-mpesa-api-using-node-js/index.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+```js
+//get access token.
+async getAccessToken(req,res,next){
+    // get data from headers by destructuring
+    let {consumerkey,consumersecret} = req.headers;
+    // request url
+    let url = "https://sandbox.safaricom.co.ke/oauth/v1/generate?grant_type=client_credentials";
+
+    // base64 encoded string from a buffer
+    let buf = new Buffer.from(`${consumerkey}:${consumersecret}`).toString("base64");
```
Comment (WanjaMIKE) [inline articles/integrating-b2c-mpesa-api-using-node-js/index.md:70]: what does the variable buf store? Could you inform the reader

File: articles/integrating-b2c-mpesa-api-using-node-js/index.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+res.status(200);
+
+// send to the client
+return res.send({
+    result:response.data
+});
+};
+```
+
+From above:
```
Comment (WanjaMIKE) [inline articles/integrating-b2c-mpesa-api-using-node-js/index.md:196]: Could you explain more from the above code

**Review by WanjaMIKE — CHANGES_REQUESTED** [review]
Nice article @Roba-W Could you review the above comments

File: articles/integrating-b2c-mpesa-api-using-node-js/index.md (hunk 1 above)
Comment (Roba-W) [reply articles/integrating-b2c-mpesa-api-using-node-js/index.md]: done.

File: articles/integrating-b2c-mpesa-api-using-node-js/index.md (hunk 2 above)
Comment (Roba-W) [reply articles/integrating-b2c-mpesa-api-using-node-js/index.md]: done.

File: articles/integrating-b2c-mpesa-api-using-node-js/index.md (hunk 3 above)
Comment (Roba-W) [reply articles/integrating-b2c-mpesa-api-using-node-js/index.md]: done.

File: articles/integrating-b2c-mpesa-api-using-node-js/index.md (hunk 4 above)
Comment (Roba-W) [reply articles/integrating-b2c-mpesa-api-using-node-js/index.md:70]: done

File: articles/integrating-b2c-mpesa-api-using-node-js/index.md (hunk 5 above)
Comment (Roba-W) [reply articles/integrating-b2c-mpesa-api-using-node-js/index.md]: done

**Comment by Roba-W** [comment]
@WanjaMIKE Have done the requested corrections

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by WanjaMIKE** [comment]
@Roba-W Do you have an author's page? I do not see you name in the `authors` folder. If you do not have one, please create it. Thanks

**Comment by hectorkambow** [comment]
cannot not merge without the author page @Roba-W

**Comment by Roba-W** [comment]
Robert is an Information Technology undergraduate at Cooperative University of Kenya. He is fluent in both web development and front and back-end using JavaScript. He is very interested in android development and Linux.
IMG_4852

**Comment by Roba-W** [comment]
@hectorkambow I have commented the author information

**Comment by WanjaMIKE** [comment]
@Roba-W As I said, you should create a `folder` in the `authors` folder and put in the above `image` and `description file`.  Just open the `authors` folder and do what other contributors have done but with your `details` and push it alongside this PR. `I hope you understand me.`

**Comment by Roba-W** [comment]
@WanjaMIKE I have made the change.

**Review by WanjaMIKE — APPROVED** [review]

**Comment by hectorkambow** [comment]
> cannot not merge without the author page @Roba-W

Oh my mistake - I did not see them Perhaps I missed them. 

Thank you for creating an author page - I'll review it now 👍

**Pull request merged**